judge/contest_format/ioi16.py

```diff
@@ -0,0 +1,98 @@
+from django.db import connection
+from django.utils.translation import gettext_lazy
+
+from judge.contest_format.ioi import IOIContestFormat
+from judge.contest_format.registry import register_contest_format
+from judge.timezone import from_database_time
+
+
+@register_contest_format('ioi16')
+class IOI16ContestFormat(IOIContestFormat):
+    name = gettext_lazy('IOI16')
+    config_defaults = {'cumtime': False}
+    '''
+        cumtime: Specify True if time penalties are to be computed. Defaults to False.
+    '''
+
+    def __init__(self, contest, config):
+        super(IOI16ContestFormat, self).__init__(contest, config)
+
+    def update_participation(self, participation):
+        cumtime = 0
+        score = 0
+        format_data = {}
+
+        with connection.cursor() as cursor:
+            cursor.execute('''
+                SELECT *
+                FROM (
+                         SELECT q.prob,
+                                q.date,
```

Comment:
The default behavior of `GROUP BY` here is actually undefined. So we need to make sure we take the minimum.
```suggestion
                                MIN(q.date) as `date`,
```